### PR TITLE
[READY] adds visible contents for smartfridges

### PR DESCRIPTION
**Pull request by Mickyan**
## Changelog
:cl:
tweak: the contents on the smartfridge icon now change depending on how many items it contains
fix: opening the maintenance panel of smartfridges now correctly updates the icon
/:cl:

Visual feedback is good

Screenshot_1

let me know if there's a more straight forward way to handle the icon updates

### CODE REVIEW

**Comment by Angustmeta** [comment]
Are items counted per stack or per individual item? Say I add 41 identical patches of styptic powder. Will it count as 41 items or 1 item?

**Comment by Mickyan** [comment]
individual, so 41

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 1]
```diff
@@ -42,11 +42,18 @@
...
 	if(!stat)
-		icon_state = startstate
+		switch(contents.len)
+			if(0)
+				icon_state = "[initial(icon_state)]"
+			if(1 to 20)
+				icon_state = "[initial(icon_state)]1"
+			if(21 to 40)
+				icon_state = "[initial(icon_state)]2"
+			if(41 to INFINITY)
```
Comment (Tlaltecuhtli) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: maybe increse these numbers? roundstart fridge already has enough items to have icon[2] and a good fridge has 200+ pills

Comment (Mickyan) [reply code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: roundstart fridge has 19 items

I can probably increase this a bit although this used for more than just chemistry and high numbers don't matter so long as you can tell the machine has been stocked

Comment (Angustmeta) [reply code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: I like it if the chem fridge does start close to the next icon threshold, so you can tell from a glance if the chemists have put anything there at all. Last stage needing more items sounds good though.

**Comment by ATH1909** [comment]
> Are items counted per stack or per individual item? Say I add 41 identical patches of styptic powder. Will it count as 41 items or 1 item?

I personally think that the former would be the better of the two options (and cause fewer issues with the roundstart things in chemistry's chemfridge already mostly filling up the smartfridge's sprite).

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm (hunk 1 above)
Comment (Tlaltecuhtli) [reply code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: roundstart should be on icon1

**Comment by ike709** [comment]
It'd be way more useful if it counted per stack.

**Comment by DaxDupont** [comment]
Yeah imagine vending machines irl with those spiral metal things

**Comment by Farquaar** [comment]
I dig it.

**Pull request merged**